File: src/content/1.8/code/scala/snippet15.scala
implicit val treeFunctor = new Functor[Tree] {
  def fmap[A, B](f: A => B)(fa: Tree[A]): Tree[B] = fa match {
    case Leaf(a) => Leaf(f(a))
    case Node(t, t1) => Node(fmap(f)(t), fmap(f)(t1))
  }
}

File: src/content/1.8/code/ocaml/snippet15.ml
module TreeFunctor : Functor with type 'a t = 'a tree = struct
  type 'a t = 'a tree

  let rec fmap f = function
    | Leaf a -> Leaf (f a)
    | Node (l, r) -> Node (fmap f l, fmap f r)
end
